# <span style="font-size:20pt;font-family:Montserrat, sans-serif;color:rgb(92, 148, 206);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Graded Assignment 3: 9 to 5</span>

<span style="font-size:10pt;font-family:&quot;Times New Roman&quot;;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">Time to show off your SQL skills! For each question, copy the SQL query you used and make note of the answer.</span>

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">The Dataset</span>

<span style="font-size:10pt;font-family:&quot;Times New Roman&quot;;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">For this assignment, you will be using the Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results which can be found on</span> [<span style="font-size:10pt;font-family:&quot;Times New Roman&quot;;color:rgb(17, 85, 204);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">Kaggle</span>](https://www.kaggle.com/bls/employment)<span style="font-size:10pt;font-family:&quot;Times New Roman&quot;;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">.</span>

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Business Issue</span>

<span style="font-size:10pt;font-family:&quot;Times New Roman&quot;;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">You are working for the Bureau of Labor Statistics with the United States government and have been approached by your boss with an important meeting request. You have been asked by your supervisor to meet with Dolly Parton whose nonprofit is looking to shed light on the state of employment in the United States. As part of the 9 to 5 project, their research is focused on production and nonsupervisory employees and how those employees fare compared to all employees in the United States. While the data the BLS collects from the CES is publicly available, Dolly Parton and her colleagues need your assistance navigating the thousands of rows in each table in LaborStatisticsDB.</span>

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">About the Dataset</span>

<span style="font-size:10pt;font-family:&quot;Times New Roman&quot;;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">This dataset comes directly from the Bureau of Labor Statistics’ Current Employment Survey (CES). Here are some things you need to know:</span>

1. <span style="font-size:10pt;background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">The industry table contains an NAICS code. This is different from the industry code. NAICS stands for North American Industry Classification System.</span>
    
2. <span style="font-size:10pt;background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">Series ID is composed of multiple different codes. CES stands for Current Employment Survey, the name of the survey which collected the data. The industry code as specified by the BLS and the data type code as specified in the datatype table.</span>
    

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Set Up</span>

<span style="font-size:10pt;font-family:&quot;Times New Roman&quot;;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">To connect to the database, use the same connection info used during the SQL lessons.&nbsp;</span> 

<span style="font-size:10pt;font-family:&quot;Times New Roman&quot;;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">For the assignment, we will be using the LaborStatisticsDB database.</span>

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Database Exploration</span>

<span style="font-size:10pt;font-family:&quot;Times New Roman&quot;;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">To start with, let’s get to know the database further.</span>

1. <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:10pt;white-space:pre-wrap;">Use this space to make note of each table in the database, the columns within each table, each column’s data type, and how the tables are connected. You can write this down or draw a diagram. Whatever method helps you get an understanding of what is going on with LaborStatisticsDB.</span>

To add a photo, diagram or document to your file, drop the file into the folder that holds this notebook.  Use the the link button to the right of the  \</\> symbol in the gray part of this cell, the link is just the name of your file.

In [ ]:
-- dbo.annual_2016

-- id

-- series_id

-- footnote_codes



-- dbo.datatype

-- data_type_code

-- data_type_text



-- dbo.footnote

-- footnote_code

-- footnote_text



-- dbo.industry

-- id

-- industry_code

-- naics_code

-- industry_name



-- dbo.january_2017

-- id

-- series_id

-- footnote_codes



-- dbo.period

-- period_code

-- month



-- dbo.seasonal

-- industry_code

-- seasonal_text



-- dbo.series

-- series_id

-- industry_code

-- series_title

-- supersector_code

-- data_type_code



-- dbo.supersector

-- supersector_code

-- supersector_name


2. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 10pt; white-space: pre-wrap;">What is the datatype for women employees?</span>

In [125]:
SELECT top 10 data_type_text
FROM LaborStatisticsDB.dbo.datatype


SELECT data_type_code
FROM LaborStatisticsDB.dbo.datatype
WHERE data_type_text = 'production and nonsupervisory employees'

(10 rows affected)

(1 row affected)

Total execution time: 00:00:00.331

data_type_text ALL EMPLOYEES AVERAGE WEEKLY HOURS OF ALL EMPLOYEES AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES AVERAGE WEEKLY OVERTIME HOURS OF ALL EMPLOYEES PRODUCTION AND NONSUPERVISORY EMPLOYEES AVERAGE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES AVERAGE WEEKLY OVERTIME HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES WOMEN EMPLOYEES AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES

data_type_code 6

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the series id for&nbsp; women employees in the commercial banking industry in the financial activities supersector?</span>

In [18]:
SELECT industry_code, industry_name
FROM LaborStatisticsDB.dbo.industry
WHERE industry_name = 'Commercial Banking'

SELECT supersector_code, supersector_name
FROM LaborStatisticsDB.dbo.supersector
WHERE supersector_name = 'Financial Activities'

SELECT series_id , series_title
FROM LaborStatisticsDB.dbo.series
WHERE supersector_code = 55 AND industry_code = 55522110 AND data_type_code = 10

(1 row affected)

(1 row affected)

(2 rows affected)

Total execution time: 00:00:00.088

industry_code industry_name 55522110 Commercial banking

supersector_code supersector_name 55 Financial activities

series_id series_title CES5552211010 Women employees CEU5552211010 Women employees

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Join in on the Fun</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to start joining! You can choose the type of join you use, just make sure to make a&nbsp; note!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join&nbsp; annual_2016 with series on series_id. We only want the data in the annual_2016 table to be included in the result.</span>

In [126]:
SELECT top 10 p.id, p.series_id, p.[year], p.[value], p.[period], p.footnote_codes, p.original_file
FROM  LaborStatisticsDB.dbo.annual_2016 as p
INNER JOIN LaborStatisticsDB.dbo.series as s
ON p.series_id = s.series_id

(10 rows affected)

Total execution time: 00:01:26.833

id series_id year value period footnote_codes original_file 8755 CEU0000000001 2016 144306 M13 NULL ce.data.0.AllCESSeries.csv 22220 CEU0000000001 2016 144306 M13 NULL ce.data.00a.TotalNonfarm.Employment.csv 8756 CEU0000000010 2016 71492 M13 NULL ce.data.0.AllCESSeries.csv 22221 CEU0000000010 2016 71492 M13 NULL ce.data.00a.TotalNonfarm.Employment.csv 6281 CEU0500000001 2016 122083 M13 NULL ce.data.05a.TotalPrivate.Employment.csv 8757 CEU0500000001 2016 122083 M13 NULL ce.data.0.AllCESSeries.csv 7954 CEU0500000002 2016 34.4 M13 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8758 CEU0500000002 2016 34.4 M13 NULL ce.data.0.AllCESSeries.csv 7955 CEU0500000003 2016 25.65 M13 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv 8759 CEU0500000003 2016 25.65 M13 NULL ce.data.0.AllCESSeries.csv

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and datatype on data_type_code</span>

In [34]:
SELECT top 20 *
FROM LaborStatisticsDB.dbo.series as s
INNER JOIN LaborStatisticsDB.dbo.datatype as d
ON s.data_type_code = d.data_type_code

(20 rows affected)

Total execution time: 00:00:00.060

series_id supersector_code industry_code data_type_code seasonal series_title data_type_code data_type_text CES0000000001 0 00000000 1 S All employees 1 ALL EMPLOYEES CES0000000010 0 00000000 10 S Women employees 10 WOMEN EMPLOYEES CES0000000025 0 00000000 25 S All employees 25 ALL EMPLOYEES CES0000000026 0 00000000 26 S All employees 26 ALL EMPLOYEES CES0500000001 5 05000000 1 S All employees 1 ALL EMPLOYEES CES0500000002 5 05000000 2 S Average weekly hours of all employees 2 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0500000003 5 05000000 3 S Average hourly earnings of all employees 3 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000006 5 05000000 6 S Production and nonsupervisory employees 6 PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 7 AVERAGE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 8 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000010 5 05000000 10 S Women employees 10 WOMEN EMPLOYEES CES0500000011 5 05000000 11 S Average weekly earnings of all employees 11 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES CES0500000012 5 05000000 12 S Average weekly earnings of all employees 12 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES CES0500000013 5 05000000 13 S Average hourly earnings of all employees 13 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000016 5 05000000 16 S Indexes of aggregate weekly hours of all employees 16 INDEXES OF AGGREGATE WEEKLY HOURS OF ALL EMPLOYEES CES0500000017 5 05000000 17 S Indexes of aggregate weekly payrolls of all employees 17 INDEXES OF AGGREGATE WEEKLY PAYROLLS OF ALL EMPLOYEES CES0500000019 5 05000000 19 S Average weekly hours of all employees 19 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0500000021 5 05000000 21 S Diffusion indexes 21 DIFFUSION INDEXES CES0500000022 5 05000000 22 S Diffusion indexes 22 DIFFUSION INDEXES CES0500000023 5 05000000 23 S Diffusion indexes 23 DIFFUSION INDEXES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and industry on industry_code</span>

In [127]:
SELECT top 10 *
FROM LaborStatisticsDB.dbo.series as s
INNER JOIN LaborStatisticsDB.dbo.industry as i
ON s.industry_code = i.industry_code

(10 rows affected)

Total execution time: 00:00:00.073

series_id supersector_code industry_code data_type_code seasonal series_title id industry_code naics_code publishing_status industry_name display_level selectable sort_sequence CES0000000001 0 00000000 1 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000010 0 00000000 10 S Women employees 0 0 - B Total nonfarm 0 T 1 CES0000000025 0 00000000 25 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000026 0 00000000 26 S All employees 0 0 - B Total nonfarm 0 T 1 CES0500000001 5 05000000 1 S All employees 1 5000000 - A Total private 1 T 2 CES0500000002 5 05000000 2 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000003 5 05000000 3 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000006 5 05000000 6 S Production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Aggregate Your Friends and Code some SQL</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Put together the following:</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; No chopping people into little bits, please.</span>

In [104]:
SELECT ROUND(SUM(p.value), 0) AS total_employees
FROM LaborStatisticsDB.dbo.annual_2016 AS p
INNER JOIN LaborStatisticsDB.dbo.series AS s
ON p.series_id = s.series_id
WHERE s.data_type_code = '1' 



(1 row affected)

Total execution time: 00:00:00.110

total_employees 2340612

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many women employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp;</span>

In [57]:
SELECT ROUND(SUM(p.value), 0) AS women_employees
FROM LaborStatisticsDB.dbo.annual_2016 AS p
INNER JOIN LaborStatisticsDB.dbo.series AS s
ON p.series_id = s.series_id
WHERE s.data_type_code = '10' AND p.year = '2016'


(1 row affected)

Total execution time: 00:00:00.069

women_employees 1125490

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many production/nonsupervisory employees were reported in 2016? Round to the nearest whole number.&nbsp;</span>

In [63]:
SELECT ROUND(SUM(p.value), 0) AS  production_nonsupervisory_employees
FROM LaborStatisticsDB.dbo.annual_2016 AS p
INNER JOIN LaborStatisticsDB.dbo.series AS s
ON p.series_id = s.series_id
WHERE s.data_type_code = '6' AND p.year = '2016'

(1 row affected)

Total execution time: 00:00:00.075

production_nonsupervisory_employees 1263650

4. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, what is the average weekly hours worked by production and nonsupervisory employees across all industries?</span>

In [70]:
SELECT ROUND(AVG(p.value), 0) AS average_weekly_hours
FROM LaborStatisticsDB.dbo.january_2017 AS p
INNER JOIN LaborStatisticsDB.dbo.series AS s
ON p.series_id = s.series_id
WHERE s.data_type_code = '7' 

(1 row affected)

Total execution time: 00:00:00.088

average_weekly_hours 36

5. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the total weekly payroll for production and nonsupervisory employees across all industries in January 2017? Round to the nearest penny, please.</span>

In [76]:
SELECT ROUND(SUM(p.value), 2) AS total_weekly_payroll
FROM LaborStatisticsDB.dbo.january_2017 AS p
INNER JOIN LaborStatisticsDB.dbo.series AS s
ON p.series_id = s.series_id
WHERE s.data_type_code = '82' AND  p.year = '2017' AND p.period = 'M01'

(1 row affected)

Total execution time: 00:00:00.086

total_weekly_payroll 1838753220

6. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, which industry was the average weekly hours worked by production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [82]:
--highest
SELECT top 10 i.industry_name,
       AVG(j.value) AS avg_weekly_hours
FROM LaborStatisticsDB.dbo.january_2017 AS j
INNER JOIN LaborStatisticsDB.dbo.series AS s 
ON j.series_id = s.series_id
INNER JOIN LaborStatisticsDB.dbo.industry AS i
ON s.industry_code = i.industry_code
WHERE s.data_type_code = '7'
  AND j.year = '2017'
GROUP BY i.industry_name
ORDER BY avg_weekly_hours DESC

--lowest
SELECT top 10 i.industry_name,
       AVG(j.value) AS avg_weekly_hours
FROM LaborStatisticsDB.dbo.january_2017 AS j
INNER JOIN LaborStatisticsDB.dbo.series AS s 
ON j.series_id = s.series_id
INNER JOIN LaborStatisticsDB.dbo.industry AS i
ON s.industry_code = i.industry_code
WHERE s.data_type_code = '7'
  AND j.year = '2017'
GROUP BY i.industry_name
ORDER BY avg_weekly_hours ASC

(10 rows affected)

(10 rows affected)

Total execution time: 00:00:00.158

industry_name avg_weekly_hours Motor vehicle power train components 49.6 Support activities for mining 47.2 Support activities for oil and gas operations 46.7 Motor vehicle metal stamping 46.699999999999996 Coal mining 46.6 Oil and gas pipeline construction 46.1 Mining 46.025 Boilers 45.7 Crushed and broken limestone mining 45.7 Mining and logging 45.65

industry_name avg_weekly_hours Fitness and recreational sports centers 16.85 Civic and social organizations 18.3 Family clothing stores 18.8 Women's clothing stores 18.8 Skiing facilities 19.3 Clothing stores 19.6 Bowling centers 20.2 Clothing accessories stores 21.2 Clothing and clothing accessories stores 21.3 Drinking places 21.35

7. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2021, which industry was the total weekly payroll for production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [105]:
--highest
SELECT top 1 i.industry_name,
       AVG(j.value) AS avg_weekly_hours
FROM LaborStatisticsDB.dbo.january_2017 AS j
INNER JOIN LaborStatisticsDB.dbo.series AS s 
ON j.series_id = s.series_id
INNER JOIN LaborStatisticsDB.dbo.industry AS i
ON s.industry_code = i.industry_code
WHERE s.data_type_code = '30'
GROUP BY i.industry_name
ORDER BY avg_weekly_hours DESC

--lowest
SELECT top 1 i.industry_name,
       AVG(j.value) AS avg_weekly_hours
FROM LaborStatisticsDB.dbo.january_2017 AS j
INNER JOIN LaborStatisticsDB.dbo.series AS s 
ON j.series_id = s.series_id
INNER JOIN LaborStatisticsDB.dbo.industry AS i
ON s.industry_code = i.industry_code
WHERE s.data_type_code = '30'
GROUP BY i.industry_name
ORDER BY avg_weekly_hours ASC


(1 row affected)

(1 row affected)

Total execution time: 00:00:00.479

industry_name avg_weekly_hours Reinsurance carriers 1810.585

industry_name avg_weekly_hours Bowling centers 234.625

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Subqueries, Unions, Derived Tables, Oh My!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Write a query that returns the series_id, industry_code, industry_name, and value from the january_2017 table but only if that value is greater than the average value for annual_2016 of data_type_code 82.</span>

In [124]:
SELECT top 10 s.series_id, i.industry_code, i.industry_name, j.[value]
FROM LaborStatisticsDB.dbo.january_2017 AS j
JOIN LaborStatisticsDB.dbo.series AS s
ON j.series_id = s.series_id
JOIN LaborStatisticsDB.dbo.industry AS i
ON s.industry_code = i.industry_code
WHERE j.value> (select avg(a.value)
     FROM LaborStatisticsDB.dbo.annual_2016 AS a
     JOIN LaborStatisticsDB.dbo.series AS s
     ON a.series_id = s.series_id
     where s.data_type_code = 82)
     ORDER BY j.value DESC;



(10 rows affected)

Total execution time: 00:00:00.479

series_id industry_code industry_name value CES0500000057 5000000 Total private 110301694 CES0500000057 5000000 Total private 110301694 CEU0500000057 5000000 Total private 109731861 CEU0500000057 5000000 Total private 109731861 CEU0800000057 8000000 Private service-providing 88446140 CEU0800000057 8000000 Private service-providing 88446140 CES0800000057 8000000 Private service-providing 88280865 CES0800000057 8000000 Private service-providing 88280865 CES0500000082 5000000 Total private 74498171 CES0500000082 5000000 Total private 74498171

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Create a Union table comparing average weekly earnings&nbsp; of production and nonsupervisory employees between annual_16 and january_17 using the data type 30.&nbsp; Round to the nearest penny.&nbsp; You should have a column for the average earnings and a column for the year, and the period.</span>

In [117]:
SELECT
    ROUND(AVG(a.value), 2) AS average_earnings,
    '2016' AS year,
    'Annual' AS period
FROM
    LaborStatisticsDB.dbo.annual_2016 AS a
JOIN LaborStatisticsDB.dbo.series AS s ON a.series_id = s.series_id
WHERE
    s.data_type_code = 30
UNION ALL
SELECT
    ROUND(AVG(j.value), 2) AS average_earnings,
    '2017' AS year,
    'January' AS period
FROM
    LaborStatisticsDB.dbo.january_2017 AS j
JOIN LaborStatisticsDB.dbo.series AS s ON j.series_id = s.series_id
WHERE
    s.data_type_code = 30;


(2 rows affected)

Total execution time: 00:00:00.109

average_earnings year period 797.2 2016 Annual 808.53 2017 January

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Summarize Your Results</span>

<span style="font-size:10pt;font-family:Arvo, serif;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">With what you know now about the </span> <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:13.3333px;white-space:pre-wrap;">Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results and working with the</span> <span style="color:rgb(102, 102, 102);white-space:pre-wrap;font-family:&quot;Times New Roman&quot;;font-size:13.3333px;">Labor Statistics Database</span><span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.</span>

1. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">During which time period did production and nonsupervisory employees fare better?</span>

Based on the data I have above that shows 797.2,  2016 Annual,

808.53, 2017, January,  the time period for January fares better.

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In which industries did production and nonsupervisory employees fare better?</span>

Motor Vehicle Power Train Components

3. <span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Now that you have explored the datasets, is there any data or information that you wish you had in this analysis</span>

<span style="font-size: 13.3333px; white-space: pre-wrap;">It was very confusing in the beginning and took way more time to understand each of the connections, they have sent the schema of the model by the time finished this project.</span>

<span style="font-size: 13.3333px; white-space: pre-wrap;">The second thing is that the instructions that are given, either are worded wrongly on specific enough because I was confused on a couple of the things because I wasn't sure which data_type-code they were referring to.</span>